openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -515,3 +573,110 @@ def partial_update(self, request):
+        https://tools.ietf.org/html/rfc7396. The content_type must be "application/merge-patch+json" or
+        else an error response with status code 415 will be returned.
+        """
+        username = request.data['username']
+        if is_username_retired(username):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        user_model = get_user_model()
+        try:
+            user = user_model.objects.get(username=username)
+
+            with transaction.atomic():
+                self.clear_pii_from_userprofile(user)
+                Registration.objects.filter(user=user).delete()
+                self.delete_users_profile_images(user)
+                self.delete_users_country_cache(user)
+                self.retire_users_data_sharing_consent(username)
+                self.retire_sapsf_data_transmission(user)
+                self.retire_user_from_pending_enterprise_customer_user(user)
+                self.retire_entitlement_support_detail(user)
+                # TODO: Password Reset links - https://openedx.atlassian.net/browse/PLAT-2104
+                # TODO: Delete OAuth2 records - https://openedx.atlassian.net/browse/EDUCATOR-2703
+                user.first_name = ''
+                user.last_name = ''
+                user.is_active = False
+                user.username = get_retired_username_by_username(user.username)
+                user.save()
+        except user_model.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        except Exception as exc:  # pylint: disable=broad-except
+            return Response(text_type(exc), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @staticmethod
+    def clear_pii_from_userprofile(user):
+        """
+        For the given user, sets all of the user's profile fields to some retired value.
+        This also deletes all ``SocialLink`` objects associated with this user's profile.
+        """
+        for model_field, value_to_assign in USER_PROFILE_PII.iteritems():
+            setattr(user.profile, model_field, value_to_assign)
+
+        user.profile.save()
+        user.profile.social_links.all().delete()
+
+    @staticmethod
+    def delete_users_profile_images(user):
+        set_has_profile_image(user.username, False)
+        names_of_profile_images = get_profile_image_names(user.username)
+        remove_profile_images(names_of_profile_images)
+
+    @staticmethod
+    def delete_users_country_cache(user):
+        cache_key = UserProfile.country_cache_key_name(user.id)
+        cache.delete(cache_key)
+
+    @staticmethod
+    def retire_users_data_sharing_consent(username):
+        DataSharingConsent.objects.filter(
+            username=username
+        ).update(
+            username=get_retired_username_by_username(username)
+        )
+
+    @staticmethod
+    def retire_sapsf_data_transmission(user):
```

Comment:
I had started a PR against the edx-enterprise Django app that would contain these removals. It'd be nice to keep this data removal in that Django app, since that's where data gets added. Thoughts?

Reply:
Yeah, Simon brought this up yesterday.  I agree it'd be nice to put it in there, but I just don't want to it, since it's already done here :) . I also don't want to delay the merging of this PR any longer than is necessary, because other work depends on it.

Reply:
My existing work (that was pretty close IIRC) is here:
https://github.com/edx/edx-enterprise/pull/306